docs/joss_paper/paper.md

```diff
@@ -0,0 +1,120 @@
+
+## Weakly-supervised learning
+Weak supervision utilising segmented corresponding regions in the medical image pairs was first proposed in a multimodal application for registering prostate MR to ultrasound images [@hu2018label]. In addition to the regularisation on the predicted displacement field, the training is driven by minimising the dissimilarity between the fixed labels and the warped moving labels, as illustrated in Fig.X. This formulation is modality-independent and similar to many other well-studied computer vision and medical imaging tasks, such as image segmentation. When multiple corresponding label pairs available, a two-stage sampling strategy can be adopted for an unbiased gradient back-propagation during training. Label dissimilarity measures include Dice, Jaccard, cross-entropy and their variants, such as a multiscale Dice.
+
+![Registration network with weak supervision loss.\label{fig:weakly}](../asset/deepreg-tutorial-weakly.svg)
+
+## Combining unsupervised loss with weak supervision
+Combining the unsupervised loss and the weak supervision has shown superior registration accuracy, compared with that using unsupervised loss alone [@balakrishnan2019voxelmorph]. As a result, the overall loss is the weighted sum of the image-, label dissimilarity and deformation regularisation.
+
+![Registration network with combined unsupervised and weakly-supervised losses.\label{fig:combined}](../asset/deepreg-tutorial-combined.svg)
+
+## Deformation regularisation and conditional segmentation
+The loss functions described in the previous sections are often combined with a deformation regularisation term on the predicted displacement field, in order to ensure the predicted deformation is smooth. The deformation can be regularised by L1-, L2 norms of the displacement gradient and bending energy, based on the first- or second dirivatives of DDF.
+
+The latest result from a different registration formulation challenged the use of deformation regularisation and consequently the necessity of predicting dense displacement field [@hu2019conditional], also implemented in DeepReg.
+
+# Implemetation
+## Model
+A pair of moving image and fixed image is the input of the network. When available, pairs of moving labels and/or fixed labels can also be the input of the network, regardless of whether they are used to compute the loss. The network predicts a dense displacement field (DDF). Options are also available for the DDF being integrated by a predicted dense velocity field (DFV) or computed from a predicted affine transformation. For conditional segmentation, the network directly predicts warped moving labels [@hu2019conditional].
+
+An encoder-decoder architecture, such as an adapted U-Net and the local-net [@hu2018weakly] is used for dense prediction such as direct DDF or DVF. An encoder-only architecture, such as ResNet and VGG, is used to predict the parameters of an affine (or rigid) transformation model, with 12 (or 6) degrees of freedom.
```

Comment:
do we have the two options for 12 and 6 degrees? i have the impression that we only have 12 DOFs